test/e2e/live_iso_test.go

```diff
@@ -0,0 +1,143 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/cluster-api/util/patch"
+
+	metal3api "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+)
+
+var _ = Describe("Live ISO", func() {
+	var (
+		specName        = "live-iso-test"
+		namespace       *corev1.Namespace
+		cancelWatches   context.CancelFunc
+		liveISOImageURL string
+		bmcUser         string
+		bmcPassword     string
+		liveISOAddress  string
+		bootMacAddress  string
+	)
+
+	BeforeEach(func() {
+		bmcUser = e2eConfig.GetVariable("BMC_USER")
+		bmcPassword = e2eConfig.GetVariable("BMC_PASSWORD")
+		bootMacAddress = e2eConfig.GetVariable("BOOT_MAC_ADDRESS")
+		liveISOImageURL = e2eConfig.GetVariable("LIVE_ISO_IMAGE")
+		liveISOAddress = e2eConfig.GetVariable("LIVE_ISO_ADDRESS")
+		namespace, cancelWatches = framework.CreateNamespaceAndWatchEvents(ctx, framework.CreateNamespaceAndWatchEventsInput{
+			Creator:   clusterProxy.GetClient(),
+			ClientSet: clusterProxy.GetClientSet(),
+			Name:      fmt.Sprintf("%s-%s", specName, util.RandomString(6)),
+			LogFolder: artifactFolder,
+		})
+	})
+
+	It("should provision a BMH with live ISO and then deprovision it", func() {
+		By("Creating a secret with BMH credentials")
+		bmcCredentials := corev1.Secret{
```

Comment:
nit: the BMC handling will happen in all tests in a similar fashion, we should probably move it to a common place

Reply:
Agree, have a look: [here](https://github.com/metal3-io/baremetal-operator/pull/1448)